### PR TITLE
Stats: Tweaks to shortcut rendering.

### PR DESCRIPTION
**Pull request by a8ck3n**
Related to #81939.

## Proposed Changes

Some clean-up under the hood.

1. Only renders the icon when needed.
2. Removes CSS rules that managed the icon visibility.
3. Removes unused class name on icons.

## Testing Instructions

1. Load the Live branch.
2. Visit the **Stats → Traffic** page.
3. Enable the date picker: `?flags=stats/date-control`
4. Confirm that selection still works as expected in the date control.
5. Confirm (via the inspector) that icons are only rendered to the DOM for the selected row.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-date-control/stats-date-control-picker-shortcuts.tsx [hunk 1]
```diff
@@ -8,11 +8,15 @@ const DateControlPickerShortcuts = ( {
...
+	// Simple selection test.
+	function isSelectedShortcut( shortcut: DateControlPickerShortcut ) {
+		return shortcut.id === currentShortcut;
+	}
 	// Apply selection state via CSS.
 	function classNameForShortcut( shortcut: DateControlPickerShortcut ): string {
 		const defaultClassName = 'date-control-picker-shortcuts__shortcut';
 		const selectedClassName = defaultClassName + ' is-selected';
-		return shortcut.id !== currentShortcut ? defaultClassName : selectedClassName;
+		return isSelectedShortcut( shortcut ) ? selectedClassName : defaultClassName;
```
Comment (kangzj) [inline client/my-sites/stats/stats-date-control/stats-date-control-picker-shortcuts.tsx:19]: This is not related to the PR and I missed the note before.

We are heavily using lib `classnames` which is super handy for scenarios like this. We could simple do the following in the tag  `<li>`:

```JSX
<li className={classNames('date-control-picker-shortcuts__shortcut', {'is-selected': isSelectedShortcut( shortcut ) })}>
```

Do you think it looks clearer?

**Review by kangzj — COMMENTED** [review]
It works as described however the styling seems to be a bit off.
|Before|After|
|----|----|
|image | image|

Left another improvement suggestion but which is not related to the PR tho.

**Review by kangzj — APPROVED** [review]
It works well after a rebase 👍

**Comment by a8ck3n** [comment]
> It works as described however the styling seems to be a bit off.

Hey @kangzj. You're actually looking at the wrong UI there. I should have included a screenshot to avoid confusion. Sorry about that.

This change is for the date picker (not the interval picker) and doesn't have any visual impact. It's strictly cleaning up the underlying HTML structure so that we only draw the icon on the row if it's selected. I'm was a bit concerned that drawing all the icons and using CSS to determine visibility might be an a11y code smell.

File: client/my-sites/stats/stats-date-control/stats-date-control-picker-shortcuts.tsx (hunk 1 above)
Comment (a8ck3n) [reply client/my-sites/stats/stats-date-control/stats-date-control-picker-shortcuts.tsx:19]: I'll have another look at this. Thanks for the reminder about the `classnames` call.
